### PR TITLE
Please add this doc in the Article section of the official rco website

### PR DESCRIPTION
**Pull request by saxenism**

### CODE REVIEW

File: vignettes/gsoc2020-final-report.Rmd [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+* [**PR #4 related to Google Code-In**](https://github.com/jcrodriguez1989/rco/pull/165)
+* [**PR #5 related to Google Code-In**](https://github.com/jcrodriguez1989/rco/pull/166)
+* [**PR #6 related to Google Code-In**](https://github.com/jcrodriguez1989/rco/pull/167)
+* [**PR #7 related to Google Code-In**](https://github.com/jcrodriguez1989/rco/pull/168)
+
+## Unexpected Turn of Events:
+
+As **Dwight D. Eisenhower** once said,
+
+> In preparing for battle I have always found that plans are useless, but planning is indispensable.
```
Comment (jcrodriguez1989) [inline vignettes/gsoc2020-final-report.Rmd]: I would put this in italic.

File: vignettes/gsoc2020-final-report.Rmd [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+
+I had listed several optimization techniques that could have been implemented in `rco` in my [*GSoC* proposal](https://drive.google.com/file/d/1HvF0oE9gW4BYt2Lo-1zRl1dcbsG1wEPg/view) as well the [*potential optimizers* vignette](https://rpubs.com/hinduBale/strategies_ideas_rco) but when we got down to discussing the design of these optimizers, it became quite clear to me and my mentors that we were walking on thin ice. With the extreme constraints that are imposed due to R's flexible nature, only a few of our optimization strategies seemed bullet-proof.
```
Comment (jcrodriguez1989) [inline vignettes/gsoc2020-final-report.Rmd]: Is it ok that _GSoC_ is italic here?

File: vignettes/gsoc2020-final-report.Rmd [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+
+I decided to tackle this bug head on. I [created a separate branch](https://github.com/hinduBale/rco/tree/issue%23107) for this bug and reproduced the bug. After lots of trials and tribulations and lots of reverse-engineering, I zeroed in one the problem. It was a problem that existed in very low-level R code. When a R script without the use of `;` is parsed, the they get tokenized as something called `expr` but when `;` was used, the parser tokenized them as an `exprlist`. The exsisting optimizer didnot handle the case of `exprlist`, so I appended code that handled `exprlist` and then the optimizer started functioning normally irrespective of the usage of `;`. This is an issue which could arise in other optimizers too, but given that, it has been solved in one optimizer, solving them in other optimizers would not prove challenging. Also, in the process of making the `opt_dead_expr()` optimizer work, I further bolstered the robustness of the optimizer by **doubling the number of test cases** that the optimizer must pass, including tests with `;`.
```
Comment (jcrodriguez1989) [inline vignettes/gsoc2020-final-report.Rmd]: typo: "the they".

Comment (jcrodriguez1989) [reply vignettes/gsoc2020-final-report.Rmd]: typo: "didnot".

File: vignettes/gsoc2020-final-report.Rmd [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+
+* [**PR for solution of issue #107**](https://github.com/jcrodriguez1989/rco/pull/170)
+
+### Call for Actions:
+
+To reduce dependencies on 3rd party applications and to promote homogeneity, we decided to go for an in-house solution for the CI/CD needs of our package, ie `Github Actions`. Earlier the popular choice for carrying out CI/CD operations were 3rd party applications such as *Travis CI* or *codecov.io*, but lately more and more developers and organizations had been migrating to Github Actions and we decided to follow suit. *Github Marketplace* wasn't much help, as there were not much support for the **R Language** as compared to more popular languages such as *Rust*, *Ruby*, *Typescript*, etc. So, I went through several documentations and scarce examples and [created a branch](https://github.com/hinduBale/rco/tree/ghActions) that [renders a website](https://hindubale.github.io/rco/) and completes the testing that Travis used to do. We left out the *code coverage* as *Github* [doesnot yet support badges that shows the percentage](https://github.com/r-lib/actions/issues/156#event-3643325125) of coverage.
```
Comment (jcrodriguez1989) [inline vignettes/gsoc2020-final-report.Rmd]: I would avoid the "more popular ...".

Comment (jcrodriguez1989) [reply vignettes/gsoc2020-final-report.Rmd]: typo: "doesnot".

File: vignettes/gsoc2020-final-report.Rmd [hunk 5]
```diff
@@ -0,0 +1,183 @@
+---
```
Comment (jcrodriguez1989) [inline vignettes/gsoc2020-final-report.Rmd:1]: Add you as an author.
Maybe add a link to your website or mail.

**Review by jcrodriguez1989 — COMMENTED** [review]
Left some comments :)

File: vignettes/gsoc2020-final-report.Rmd (hunk 1 above)
Comment (saxenism) [reply vignettes/gsoc2020-final-report.Rmd]: The quote or the author 😅

**Review by jcrodriguez1989 — APPROVED** [review]
Great work Rahul!

**Pull request merged**